lib/yard/parser/ruby/ruby_parser.rb

```diff
@@ -656,10 +656,12 @@ def add_comment(line, node = nil, before_node = nil, into = false)
         end
 
         def freeze_tree(node = nil)
-          node ||= root
-          node.children.each do |child|
-            child.parent = node
-            freeze_tree(child)
+          nodes = []
+          nodes.push(node || root)
```

Comment:
Instead of the separate push call, this should just be `nodes = [node || root]`.